libraries/AP_Airspeed/AP_Airspeed.cpp

```diff
@@ -399,16 +436,137 @@ void AP_Airspeed::read(void)
     }
 #endif
 
-    // setup primary
-    if (healthy(primary_sensor.get())) {
-        primary = primary_sensor.get();
-        return;
-    }
-    for (uint8_t i=0; i<AIRSPEED_MAX_SENSORS; i++) {
-        if (healthy(i)) {
-            primary = i;
+    switch(blend_conf) {
```

Comment:
this switch of 2 items seems overkill since one case is trivial and the other very complicated. Might as well be:
```
if (blend_conf == BLEND_NONE ||
  !_ahrs->have_inertial_nav() ||
  !_ahrs->get_velocity_NED(velocity)) {
    process_no_blend();
    return;
}
.. more stuff
```

Reply:
might be...